src/screens/Forum/ForumCategoriesScreen.js

```diff
@@ -1,10 +1,60 @@
-import React from 'react';
-import { View, Text } from 'react-native';
+import React, { useState, useEffect } from 'react';
+import {
+  ScrollView, Text, View, StyleSheet,
+} from 'react-native';
+import firestore from '@react-native-firebase/firestore';
+import PropTypes from 'prop-types';
+import ForumCategoryCard from './ForumCategoryCard';
+
+
+const styles = StyleSheet.create({
+  container: {
+    padding: 10,
+    backgroundColor: '#daecf5',
+  },
+});
+
+export default function ForumCategoriesScreen({ navigation }) {
+  const [errorMessage, setErrorMessage] = useState(null);
+  const [forumCategories, setForumCategories] = useState([]);
+
+  useEffect(() => { /* Put firestore query inside useEffect hook to prevent infinite loop */
+    firestore()
+      .collection('forum_categories')
+      .orderBy('title')
+      .get()
+      .then((snapshot) => {
+        const forumCategoriesData = snapshot.docs.map((doc) => ({
+          ...doc.data(),
+          id: doc.id,
+        }));
+        setForumCategories(
+          forumCategoriesData
+            .map((forum) => {
+              const navigateToSubcategory = () => navigation.navigate('ForumSubcategoryPosts', { categoryID: forum.id });
+              return (
+                <ForumCategoryCard key={forum.id} navigate={navigateToSubcategory}>
+                  {forum.title}
+                </ForumCategoryCard>
+              );
+            }),
+        );
+      })
+      .catch((error) => {
+        setErrorMessage(error.message);
+      });
+  });
```

Comment:
The infinite loop is still here. We have to add the second argument to `useEffect`:
```
useEffect(() = {
  ...
}, dependencyList)
```
where `dependencyList` is an array. If you just put an empty array, eslint should automatically fill the array with whatever dependencies your effect has (if it has any).